### PR TITLE
WRKLDS-1449: add ObjectMeta to authorization and security types

### PR DESCRIPTION
**Pull request by atiratree**

### CODE REVIEW

File: authorization/v1/types.go [hunk 1]
```diff
@@ -232,7 +240,7 @@ type ResourceAccessReviewResponse struct {
 
 // +genclient
 // +genclient:nonNamespaced
-// +genclient:skipVerbs=apply,get,list,create,update,patch,delete,deleteCollection,watch
+// +genclient:skipVerbs=apply,applyStatus,get,list,create,update,updateStatus,patch,delete,deleteCollection,watch
```
Comment (deads2k) [inline authorization/v1/types.go:243]: onlyVerb create?

File: authorization/v1/types.go [hunk 2]
```diff
@@ -280,7 +292,7 @@ func (t OptionalScopes) String() string {
...
```
Comment (deads2k) [inline authorization/v1/types.go:295]: only verb create?

File: authorization/v1/types.go [hunk 3]
```diff
@@ -306,7 +322,7 @@ type SubjectAccessReview struct {
 }
 
 // +genclient
-// +genclient:skipVerbs=apply,get,list,create,update,patch,delete,deleteCollection,watch
+// +genclient:skipVerbs=apply,applyStatus,get,list,create,update,updateStatus,patch,delete,deleteCollection,watch
```
Comment (deads2k) [inline authorization/v1/types.go:325]: onlyverb create?

File: authorization/v1/types.go [hunk 4]
```diff
@@ -317,12 +333,16 @@ type SubjectAccessReview struct {
...
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,2,opt,name=metadata"`
+
+	// Action describes the action being tested. The Namespace element is FORCED to the current namespace.
 	Action `json:",inline" protobuf:"bytes,1,opt,name=Action"`
 }
 
 // +genclient
-// +genclient:skipVerbs=apply,get,list,create,update,patch,delete,deleteCollection,watch
+// +genclient:skipVerbs=apply,applyStatus,get,list,create,update,updateStatus,patch,delete,deleteCollection,watch
```
Comment (deads2k) [inline authorization/v1/types.go:345]: onlyverb create

**Comment by deads2k** [comment]
good in concept.  I suggest switch to the only support verbs.  I'll approve for you to merge once the client bits are worked out

/approve

File: authorization/v1/types.go (hunk 1 above)
Comment (atiratree) [reply authorization/v1/types.go:243]: I think it conflicts with

```
// +genclient:method=Create,verb=create,result=ResourceAccessReviewResponse
```

Comment (atiratree) [reply authorization/v1/types.go:243]: generate create twice

```

// Create takes the representation of a resourceAccessReview and creates it.  Returns the server's representation of the resourceAccessReview, and an error, if there is any.
func (c *resourceAccessReviews) Create(ctx context.Context, resourceAccessReview *v1.ResourceAccessReview, opts metav1.CreateOptions) (result *v1.ResourceAccessReview, err error) {
	result = &v1.ResourceAccessReview{}
	err = c.client.Post().
		Resource("resourceaccessreviews").
		VersionedParams(&opts, scheme.ParameterCodec).
		Body(resourceAccessReview).
		Do(ctx).
		Into(result)
	return
}

// Create takes the representation of a resourceAccessReview and creates it.  Returns the server's representation of the resourceAccessReviewResponse, and an error, if there is any.
func (c *resourceAccessReviews) Create(ctx context.Context, resourceAccessReview *v1.ResourceAccessReview, opts metav1.CreateOptions) (result *v1.ResourceAccessReviewResponse, err error) {
	result = &v1.ResourceAccessReviewResponse{}
	err = c.client.Post().
		Resource("resourceaccessreviews").
		VersionedParams(&opts, scheme.ParameterCodec).
		Body(resourceAccessReview).
		Do(ctx).
		Into(result)
	return
}

```

**Review by bertinatto — COMMENTED** [review]
/lgtm

**Pull request merged**